src/modules/follow_target_estimator/TargetEstimator.hpp

```diff
@@ -0,0 +1,213 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file TargetEstimator.hpp
+ *
+ * @author Alessandro Simovic <potaito-dev@protonmail.com>
+ */
+
+#pragma once
+
+#include <uORB/SubscriptionInterval.hpp>
+#include <uORB/topics/follow_target.h>
+#include <uORB/topics/vehicle_local_position.h>
+#include <uORB/topics/follow_target_estimator.h>
+#include <uORB/topics/parameter_update.h>
+#include <uORB/PublicationMulti.hpp>
+
+#include <drivers/drv_hrt.h>
+#include <lib/mathlib/math/filter/AlphaFilter.hpp>
+#include <lib/geo/geo.h>
+#include <mathlib/mathlib.h>
+#include <px4_platform_common/module.h>
+#include <px4_platform_common/module_params.h>
+
+#define GPS_MESSAGE_STALE_TIMEOUT_MS 3000  	// Duration after which the connection to the target is considered lost
```

Comment:
nitpick, use `static constexpr TYPE` instead of defines.

Reply:
Done. But when using `constexpr`, I should either move the variables to the .cpp file, or put them inside a namespace in the .hpp file, right?

Reply:
My preference is contained in the class, but for these simple constants you can put them wherever you think it makes sense.